dashboard/app/assets/javascripts/header.js

```diff
@@ -56,7 +53,7 @@ dashboard.buildHeader = function (stageData, progressData, currentLevelId, userI
   }
   var progressContainer = $('.progress_container');
   stageData.levels.forEach(function(level, index, levels) {
-    var status = (levelProgress[level.id] || {}).status || 'not_tried';
+    var status = mergeProgress((progressData.levels[level.id] || {}).result, clientLevelProgress[level.id]);
```

Comment:
I believe clientLevelProgress can be null here, so we need to defend against that.